metta/reduct/rules.metta

```diff
@@ -0,0 +1,59 @@
+(: chain-reduction-rule (-> Atom))
+(: recurse-reduction-rule (-> Atom))
+(: commutative-reduction-rule (-> Atom))
+
+(= (chain-reduction-rule)
+   (: ChainReduction
+      (-> (examplar $a)
+          (reduced $a $b)
+          (examplar $b)))
```

Comment:
I don't think this is correct, not in the original MOSES implementation anyway.  First, reduction alone is not sufficient to define if a program is an exemplar.  And second, that `$a` is an exemplar is in the premise, which could be even contradictory if the user wants only reduced candidates as exemplar.  I think you should leave the notion of exemplar completely out of the reduct engine.  After all it will be useful in order applications than MOSES.